codec-http/src/main/java/io/netty/handler/codec/http/HttpRequestDecoder.java

```diff
@@ -320,8 +322,46 @@ private static boolean isPostMethod(final byte[] sb, int start) {
         return maybePost == POST_AS_INT;
     }
 
+    private static int firstInvalidMethodOctet(final byte[] sb, int start, int length) {
+        int end = start + length;
+        for (int i = start; i < end; i++) {
+            if (sb[i] < 0 || !VALID_HTTP_METHOD_OCTETS.get(sb[i])) {
+                return i;
+            }
+        }
+        return -1;
+    }
+
+    private static final BitSet VALID_HTTP_METHOD_OCTETS = validHttpMethodOctets();
+
+    private static BitSet validHttpMethodOctets() {
+        BitSet bits = new BitSet();
+        // https://datatracker.ietf.org/doc/html/rfc9110#name-tokens
+        // token          = 1*tchar
+        // tchar          = "!" / "#" / "$" / "%" / "&" / "'" / "*"
+        //                 / "+" / "-" / "." / "^" / "_" / "`" / "|" / "~"
+        //                 / DIGIT / ALPHA
+        //                 ; any VCHAR, except delimiters
+        //Many HTTP field values are defined using common syntax components,
+        // separated by whitespace or specific delimiting characters. Delimiters
+        // are chosen from the set of US-ASCII visual characters not allowed in
+        // a token (DQUOTE and "(),/:;<=>?@[\]{}").
+        //VCHAR          =  %x21-7E
+        bits.set(0x21, 0x7f);
+        int[] delimiters = { '(', ')', ',', '/', ':', ';', '<', '=', '>', '?', '@', '"', '[', '\\', ']', '{', '}' };
+        for (int delimiter : delimiters) {
+            bits.set(delimiter, false);
+        }
+        return bits;
+    }
+
     @Override
     protected String splitFirstWordInitialLine(final byte[] sb, final int start, final int length) {
+        int firstInvalidOctet = firstInvalidMethodOctet(sb, start, length);
```

Comment:
Another approach could be to move this validation to `HttpMethod` constructor. It already has some validation. If it's not sufficient it could be improved. Advantages:
1. This will help to avoid validation here for the fast path of recognizing `GET/POST` methods.
2. It will cover not only incoming but also outgoing http methods made by netty clients.
3. It will allow you to reuse existing code for token validation from `HttpHeaderValidationUtil.validateToken`.

Reply:
Thank you for the suggestion! It is an appropriate approach. However, regarding point 3, I have a question: Is it allowed to call HttpHeaderValidationUtil.validateToken within HttpMethod? I understand that HttpMethod and HttpHeader are not the same concept.

Reply:
@freeyob I like the suggestion of @idelpivnitskiy ... At worse you could just move the `validateToken(...)` method to `HttpUtil` or so and re-use it from everywhere.